ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -223,32 +223,17 @@ message DisclosedContract {
   // Required
   oneof arguments {
     // The contract arguments as typed Record
-    Record create_arguments = 3;
+    // Deprecated in favor of `contract_create_data`
+    Record create_arguments = 3 [deprecated = true];
 
-    // The contract arguments specified using an opaque blob extracted from the ``create_arguments_blob`` field
-    // of a ``com.daml.ledger.api.v1.CreatedEvent``.
-    google.protobuf.Any create_arguments_blob = 5;
+    // Opaque byte string containing the complete payload required by the Daml engine
+    // to reconstruct a contract not known to the receiving participant.
+    // Required
+    google.protobuf.Any contract_create_payload = 5;
```

Comment:
Maybe adding a new field at the end would be preferable thus getting rid of the need for the oneOf in one stroke?

Reply:
as stated below, I'd suggest you minimize the amount of polish you do for 2.8 and instead just focus on getting the code with minimal delivery risk to a state where we can release.

IMHO, that means leaving the one_of and the field names and just deprecating the `create_arguments` field.

Reply:
I hear you, I will let the others comment before I make the change though, so that they can also see the proposal in its original form.

Reply:
The comment about this being an opaque blob and then specifying `google.protobuf.Any` sounds weird. @remyhaemmerle-da is rolling his own versioning scheme in https://github.com/digital-asset/daml/pull/17598 anyway. So what's the point of the `Any` here? It just constraints us again w.r.t. what we can do on the refactoring side.

I'm with Marcin here: Add a new field with the right name and type `bytes` at the end, and deprecate both fields in the one-of.

Reply:
`Any` is the reflection of the fact that we used VersionedBlob message in this location up until now. In the light of @remyhaemmerle-da 's recent changes, this simply needs to be a blob. A new field is thus a must

Reply:
This `Any` -> `bytes` transformation has forced me to introduce completely new fields. It is actually better this way, because we maintain formal backwards compatibility with 2.7.x